### DISCUSSION TITLE
WebRTC stream stutters

### DISCUSSION THREAD

**Discussion by TheLion092** [bluenviron/mediamtx#1691] [Questions]
I was searching for a way to transmit a stream via OBS with the lowest possible latency over the internet and integrate it as a source into another OBS.

I am streaming with OBS (RTMP) over the internet to the mediamtx server. From there, I pull the stream over WebRTC in the same network and integrate it as a browser source into another OBS. Thanks to this fantastic server, it works great, and I have a latency of about 300-500 ms.

Unfortunately, the WebRTC stream is stuttering. Every few 20-30 seconds, the stream freezes for about 2-3 seconds and then resumes. What settings can I adjust to achieve a seamless stream while maintaining the lowest possible latency (maximum 500ms)?

If I watch the stream with RTSP, I don't have any stutters, but unfortunately there is a high latency of about 2 seconds.

i already tried to play around with the OBS settings but nothing changed.
It's important for me, that there is no buffering which will increase the delay every stutter.
The server and the pull-client are in the same network with best connectivity (It even happens on the same machine).

Here are my OBS and server settings:

OBS:

```
server: rtmp://IPADDRESS
NVIDIA NVENC H.264 (on RTX3080)
CBR
12000 Kbps
Keyframe: 0
good Quality
Extreme low latency
Lookahead: disabled
B-frames: 0
```

server:

```
###############################################
# General parameters

logLevel: info
logDestinations: [stdout]
logFile: mediamtx.log

readTimeout: 10s
writeTimeout: 10s
readBufferCount: 4096
udpMaxPayloadSize: 1472

rtspDisable: yes
hlsDisable: yes


###############################################
# RTMP parameters


rtmpDisable: no
rtmpAddress: :1935
rtmpEncryption: "no"
rtmpsAddress: :1936
rtmpServerKey: server.key
rtmpServerCert: server.crt

###############################################
# WebRTC parameters


webrtcDisable: no
webrtcAddress: :8889
webrtcEncryption: no
webrtcAllowOrigin: '*'
webrtcICEServers: [stun:stun.l.google.com:19302]
```

**Comment by aler9** [comment]
Hello,

Regarding the OBS -> WebRTC stuttering, i'm investigating.

Regarding RTSP, keep in mind that WebRTC and RTSP works in the same exact way when streaming the video (they use RTP packets). Those 2 seconds of latency are introduced by VLC and can be removed by following instructions here:
https://github.com/bluenviron/mediamtx#decrease-latency

**Reply by alpe12** [reply]
In my case, RTSP works. But WebRTC freezes at least once a second. And it freezes on all WebRTC clients at the same time, so it's not a client issue. No errors on log.

PS: If I enable HLS I get "muxer error: DTS is not monotonically increasing" when the HLS stream freezes. Maybe the same issue happens silently in WebRTC? https://github.com/bluenviron/mediamtx/issues/1002

[a comment by an automated account is not shown]
